### PR TITLE
MoreJPEG command

### LINKED ISSUE #397: "Needs more JPEG" command
As suggested by @LBlend in #381.

### PR DESCRIPTION
**Pull request by Lireoy**
Closes #397 
Direct image sending and links are supported.

image
image

### CODE REVIEW

File: src/commands/image-manipulation/morejpeg.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    this.category = 'images'
+    this.parameters = new CommandParameters(this,
+      new StringParameter({ required: false })
+    )
+    this.requirements = new CommandRequirements(this, { canvasOnly: true })
+  }
+
+  async run ({ t, author, channel, message }, text) {
+    channel.startTyping()
+    if (message.attachments.first() === undefined && text === undefined) {
```
Comment (pedrofracassi) [inline src/commands/image-manipulation/morejpeg.js]:
```suggestion
    if (!message.attachments.first() && !text) {
```

File: src/commands/image-manipulation/morejpeg.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      channel.stopTyping()
+      throw new CommandError(t('commands:morejpeg.missingImage'), true)
+    } else if (message.attachments.first() !== undefined) {
```
Comment (pedrofracassi) [inline src/commands/image-manipulation/morejpeg.js]:
```suggestion
    } else if (message.attachments.first()) {
```

File: src/locales/en-US/commands.json [hunk 3]
```diff
@@ -1073,5 +1073,10 @@
     "noMember": "You need to mention someone!",
     "cantWelcomeYourself": "You can't welcome yourself!",
     "welcomeMessage": "Welcome to the team {{member}}!\nIn <#447005926981894144> you can talk about translation related stuff.\nYou will see announcements in <#476110480377905163>.\nYou will get pinged in <#450050899197296651> when there are new strings.\nLast but not least, you can see some useful info in <#447962790918094878>.\n\nIf you have any questions, feel free to ask!"
+  },
+  "morejpeg": {
+	"commandDescription": "Jpegifies the image.",
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
	"commandDescription": "Adds JPEG compression to an image.",
```

File: src/utils/CanvasTemplates.js [hunk 4]
```diff
@@ -151,7 +151,7 @@ module.exports = class CanvasTemplates {
     )
 
     // Image handling
-    const [ avatarImage, coinsImage, repImage, backgroundImage ] = await IMAGE_ASSETS
+    const [avatarImage, coinsImage, repImage, backgroundImage] = await IMAGE_ASSETS
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 5]
```diff
@@ -245,7 +245,7 @@ module.exports = class CanvasTemplates {
     ctx.writeParagraph(song.title, TITLE_FONT, LEFT_TEXT_MARGIN, TITLE_Y, RIGHT_TEXT_MARGIN, TITLE_Y + 1, 5, ALIGN.BOTTOM_LEFT)
 
     // Image handling
-    const [ mainImage, backgroundImage ] = await IMAGE_ASSETS
+    const [mainImage, backgroundImage] = await IMAGE_ASSETS
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 6]
```diff
@@ -386,11 +386,11 @@ module.exports = class CanvasTemplates {
     })
 
     // Image handling
-    const [ valueImage, medalImage, backgroundImage, ...avatarImages ] = await IMAGE_ASSETS
+    const [valueImage, medalImage, backgroundImage, ...avatarImages] = await IMAGE_ASSETS
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 7]
```diff
@@ -386,11 +386,11 @@ module.exports = class CanvasTemplates {
...
 
     const avatarImage = avatarImages.shift()
     //   Top avatar shadow
-    const [ TOP_USER_AVATAR_X, TOP_USER_AVATAR_Y ] = TOP_USER_AVATAR_COORDS
+    const [TOP_USER_AVATAR_X, TOP_USER_AVATAR_Y] = TOP_USER_AVATAR_COORDS
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 8]
```diff
@@ -409,7 +409,7 @@ module.exports = class CanvasTemplates {
 
     // Others avatars
     avatarImages.forEach((avatar, i) => {
-      const [ avatarX, avatarY ] = avatarCoords[i]
+      const [avatarX, avatarY] = avatarCoords[i]
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 9]
```diff
@@ -471,7 +471,7 @@ module.exports = class CanvasTemplates {
     const canvas = createCanvas(WIDTH, HEIGHT)
     const ctx = canvas.getContext('2d')
 
-    const [ triggeredLabel, avatarImage ] = await IMAGE_ASSETS
+    const [triggeredLabel, avatarImage] = await IMAGE_ASSETS
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 10]
```diff
@@ -527,7 +527,7 @@ module.exports = class CanvasTemplates {
     const usedIcons = daily.map(d => icon(d.icon)).reduce((a, i) => {
       if (!a.includes(i)) a.push(i)
       return a
-    }, [ icon(now.icon) ])
+    }, [icon(now.icon)])
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 11]
```diff
@@ -593,7 +593,7 @@ module.exports = class CanvasTemplates {
       ctx.drawIcon(getIcon(day.icon), day.iconX, day.iconY, DAY_ICON_SIZE, DAY_ICON_SIZE, '#fff')
     })
 
-    const [ backgroundImage, arrowImage, windImage ] = await IMAGE_ASSETS
+    const [backgroundImage, arrowImage, windImage] = await IMAGE_ASSETS
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

File: src/utils/CanvasTemplates.js [hunk 12]
```diff
@@ -720,7 +720,7 @@ module.exports = class CanvasTemplates {
     })
 
     //  IMAGES
-    const [ heartIcon, ...avatarImages ] = await IMAGE_ASSETS
+    const [heartIcon, ...avatarImages] = await IMAGE_ASSETS
```
Comment (pedrofracassi) [inline src/utils/CanvasTemplates.js]: Why?

Comment (Lireoy) [reply src/utils/CanvasTemplates.js]: All of these `Why?` modifications are Weebstorm autoformats. I haven't touched them. Shall I remove the modifications? @pedrofracassi

Comment (pedrofracassi) [reply src/utils/CanvasTemplates.js]: Please do.

**Comment by pedrofracassi** [comment]
LGTM

File: src/commands/image-manipulation/morejpeg.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
+const { CanvasTemplates, CommandStructures } = require('../../')
+const { Command, CommandRequirements, CommandParameters, StringParameter, CommandError } = CommandStructures
+
+const { Attachment } = require('discord.js')
+
+module.exports = class MoreJpeg extends Command {
+  constructor (client) {
+    super(client)
+    this.name = 'morejpeg'
+    this.aliases = ['needsmorejpeg', 'needsmorejpg', 'jpg', 'pixelate', 'compress']
```
Comment (pedrofracassi) [inline src/commands/image-manipulation/morejpeg.js]:
```suggestion
    this.aliases = ['needsmorejpeg', 'needsmorejpg', 'jpg', 'compress']
```

**Pull request merged**